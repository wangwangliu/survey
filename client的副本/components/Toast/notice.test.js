/* eslint-disable no-undef */
import "babel-polyfill";
import React from 'react';
import Notice from './Notice';

jest.useFakeTimers();
test('toast component animation', async () => {
  const notice = shallow(<Notice />);
  const instance = notice.instance();
  jest.runAllTimers();
  notice.update();
  expect(notice).toMatchSnapshot();
  await instance.componentWillUnmount();
  expect(notice).toMatchSnapshot();
});
test('toast component with content', () => {
  const notice = shallow(<Notice content="test content" />);
  expect(notice).toMatchSnapshot();
});
test('toast component with nomask', () => {
  const notice = shallow(<Notice mask={false} />);
  expect(notice).toMatchSnapshot();
});
