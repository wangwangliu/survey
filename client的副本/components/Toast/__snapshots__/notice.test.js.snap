// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`toast component animation 1`] = `
<div
  className="window mask"
>
  <div
    className="content fade-in"
  />
</div>
`;

exports[`toast component animation 2`] = `
<div
  className="window mask"
>
  <div
    className="content fade"
  />
</div>
`;

exports[`toast component with content 1`] = `
<div
  className="window mask"
>
  <div
    className="content fade"
  >
    test content
  </div>
</div>
`;

exports[`toast component with nomask 1`] = `
<div
  className="window nomask"
>
  <div
    className="content fade"
  />
</div>
`;
